annotator/src/main/java/org/cbioportal/models/MutationRecord.java

```diff
@@ -28,500 +28,87 @@
-
-    public void setVALIDATION_METHOD(String validationMethod) {
-        this.validationMethod = validationMethod;
-    }
-
-    public String getSCORE() {
-        return this.score == null ? "" : this.score;
-    }
-
-    public void setSCORE(String score) {
-        this.score = score;
-    }
-
-    public String getBAM_FILE() {
-        return this.bamFile == null ? "" : this.bamFile;
-    }
-
-    public void setBAM_FILE(String bamFile) {
-        this.bamFile = bamFile;
-    }
-
-    public String getSEQUENCER() {
-        return this.sequencer == null ? "" : this.sequencer;
-    }
-
-    public void setSEQUENCER(String sequencer) {
-        this.sequencer = sequencer;
-    }
-
-    public String getTUMOR_SAMPLE_UUID() {
-        return this.tumorSampleUUID == null ? "" : this.tumorSampleUUID;
-    }
-
-    public void setTUMOR_SAMPLE_UUID(String tumorSampleUUID) {
-        this.tumorSampleUUID = tumorSampleUUID;
-    }
-
-    public String getMATCHED_NORM_SAMPLE_UUID() {
-        return this.matchedNormSampleUUID == null ? "" : this.matchedNormSampleUUID;
-    }
-
-    public void setMATCHED_NORM_SAMPLE_UUID(String matchedNormSampleUUID) {
-        this.matchedNormSampleUUID = matchedNormSampleUUID;
-    }
-
-    public String getT_REF_COUNT() {
-        return this.tRefCount == null ? "" : this.tRefCount;
-    }
-
-    public void setT_REF_COUNT(String tRefCount) {
-        this.tRefCount = tRefCount;
-    }
-
-    public String getT_ALT_COUNT() {
-        return this.tAltCount == null ? "" : this.tAltCount;
-    }
-
-    public void setT_ALT_COUNT(String tAltCount) {
-        this.tAltCount = tAltCount;
-    }
-
-    public String getN_REF_COUNT() {
-        return this.nRefCount == null ? "" : this.nRefCount;
-    }
-
-    public void setN_REF_COUNT(String nRefCount) {
-        this.nRefCount = nRefCount;
-    }
-
-    public String getN_ALT_COUNT() {
-        return this.nAltCount == null ? "" : this.nAltCount;
-    }
-
-    public void setN_ALT_COUNT(String nAltCount) {
-        this.nAltCount = nAltCount;
-    }
-
-    public void addAdditionalProperty(String property, String value) {
-        this.additionalProperties.put(property, value);
-    }
-
-    public Map<String, String> getAdditionalProperties() {
-        return this.additionalProperties;
-    }
-
-    public void setAdditionalProperties(Map<String, String> additionalProperties) {
-        this.additionalProperties = additionalProperties;
-    }
-
-    public List<String> getHeaderWithAdditionalFields() {
-        List<String> headerWithAdditionalFields = new ArrayList<>(getHeader());
-        headerWithAdditionalFields.addAll(getSortedAdditionalPropertiesKeys());
-        return headerWithAdditionalFields;
+    // value is the header index
+    private static final Map<String, Integer> headers = new HashMap<>();
+    private static final String separator = "\t";
+    private final int[] beginIndices;
+    private final int[] endIndices;
+    private final String line;
```

Comment:
Here, rather than storing the fields separately we are storing a raw string which represents the record. It does not feel like we are modeling the data here .. because in this representation there are no assumptions about fields which are expected, must be present or may be optional, or any other semantics. This model could be used for any tab-delimited file format (like CNA or whatever) and it does not document to the programmer what the expectations are. With this code I could not come to the MutationRecord.java file in order to see what "belongs" in a MutationRecord easily. I'm not sure what else to say here ... this is big departure from our normal approach of having the model class be a representation of the expected elements of the object being modeled. We should probably discuss this.